### PR TITLE
YugabyteDB: Add resource limits to YugabyteDB pods

### PR DESCRIPTION
**Pull request by samkulkarni20**
YugabyteDB operator used to create Master & TServer pods without any resource requests/limits which would cause it to run into soft/hard memory limit issue. This PR fixes it by adding the recommended CPU and Memory requests and limits as defaults. The resource requirements can be overridden through the `ybcluster` CRD for cases when the cluster isn't big enough to provide the requested resources.

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves # https://github.com/yugabyte/yugabyte-db/issues/3884

**Checklist:**

- [X] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [X] Documentation has been updated, if necessary.
- [X] Unit tests have been added, if necessary.
- [X] Integration tests have been added, if necessary.
- [X] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [X] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test yugabytedb]

### CODE REVIEW

File: pkg/operator/yugabytedb/controller.go [hunk 1]
```diff
@@ -340,10 +373,19 @@ func createMasterContainerCommand(namespace, serviceName string, masterCompleteN
 		fmt.Sprintf("--replication_factor=%d", replicas),
 		"--logtostderr",
 	}
+
+	if &resources.Limits != nil {
+		memLimit, ok := getMemoryLimitBytes(resources)
```
Comment (travisn) [inline pkg/operator/yugabytedb/controller.go]: nit: in general it's preferred to check for ok on the same line, such as the following.
```
if memLimit, ok := getMemoryLimitBytes(resources); ok {
...
}
```

Or if on a separate line, no need for a blank line between. Not critical though if you prefer it this way.

File: Documentation/yugabytedb-cluster-crd.md [hunk 2]
```diff
@@ -78,6 +94,11 @@ Specify Master/TServer specific attributes under `master`/`tserver`. The valid a
 
 Specify replica count for `master` & `tserver` pods under `replicas` field. This is a **required** field.
 
+### Resource
+
+Specify resource requests and limits for CPU & Memory. If provided, the given resource values will be used. If omitted the default CPU request & limit will be 2, whereas default memory request & limit will be 2Gi & 4Gi for Master & TServer, respectively. You may override these values for dev environments where the cluster doesn't have CPU/Memory as per the defaults.
+**Though It is recommended to use the defaults for production-like deployments.**
```
Comment (travisn) [inline Documentation/yugabytedb-cluster-crd.md]: nit: don't capitalize "It"

File: Documentation/yugabytedb-cluster-crd.md [hunk 3]
```diff
@@ -78,6 +94,11 @@ Specify Master/TServer specific attributes under `master`/`tserver`. The valid a
 
 Specify replica count for `master` & `tserver` pods under `replicas` field. This is a **required** field.
 
+### Resource
+
+Specify resource requests and limits for CPU & Memory. If provided, the given resource values will be used. If omitted the default CPU request & limit will be 2, whereas default memory request & limit will be 2Gi & 4Gi for Master & TServer, respectively. You may override these values for dev environments where the cluster doesn't have CPU/Memory as per the defaults.
```
Comment (travisn) [inline Documentation/yugabytedb-cluster-crd.md]: Production scenarios all have the same requirements? It doesn't depend on the workload?

File: pkg/operator/yugabytedb/controller.go (hunk 1 above)
Comment (samkulkarni20) [reply pkg/operator/yugabytedb/controller.go]: Updated

File: Documentation/yugabytedb-cluster-crd.md (hunk 2 above)
Comment (samkulkarni20) [reply Documentation/yugabytedb-cluster-crd.md]: Updated

**Pull request merged**